### PR TITLE
Rename to io.jaegertracing

### LINKED ISSUE #287: Change maven group to io.jaegertracing
As discussed on friday's call, the maven group should be changed from `com.uber.jaeger` to `io.jaegertracing`. The proposal was to make this change in the `v0.22.0` branch so that it gets merged to master as part of the OT Java 0.31.0 change. 

However the implication is that the Java package would also change (as it usually follows the maven group), and this would be a breaking change - so wanted to make sure this is clear (as wasn't explicitly discussed)?

Other option would be to make this change when moving to version 1,0.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #287 #182 
Depends on https://github.com/jaegertracing/jaeger-idl/pull/47 

- Removed context, instrumentation modules, b3
- renamed to io.jaegertracing
- fixed import checkstyle

### CODE REVIEW

**Comment by pavolloffay** [comment]
This s ready for review. Lagacy repository with com.uber is here https://github.com/jaegertracing/legacy-client-java

File: jaeger-thrift/src/main/java/io/jaegertracing/thrift/reporters/protocols/ThriftUdpTransport.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
+/*
```
Comment (objectiser) [inline jaeger-thrift/src/main/java/io/jaegertracing/thrift/reporters/protocols/ThriftUdpTransport.java]: Seems like something wrong here - this class was moved from jaeger-core, but looks like the old version is still there in this PR.

File: jaeger-b3/README.md [hunk 2]
```diff
@@ -1,16 +0,0 @@
-# jaeger-b3
-These components allow you to join traces with other Zipkin (B3)
-instrumented applications (such as Brave or zipkin-go-opentracing).
-
-# Usage
```
Comment (objectiser) [inline jaeger-b3/README.md]: This section will need to be moved to the jaeger-core readme.

Comment (pavolloffay) [reply jaeger-b3/README.md]: done

**Comment by pavolloffay** [comment]
changes applied

**Comment by yurishkuro** [comment]
feel free to merge when ready, I am going to trust Gary's review

**Comment by pavolloffay** [comment]
thanks, I will merge then

**Pull request merged**